teamcity: failed tests on master: test/TestReplicateRemovedNodeDisruptiveElection

The following tests appear to have failed:

[#240426](https://teamcity.cockroachdb.com/viewLog.html?buildId=240426):

```
--- FAIL: test/TestReplicateRemovedNodeDisruptiveElection (45.900s)
soon.go:43: storage/client_raft_test.go:2494, condition failed to evaluate within 45s: expected [0 5 5 5], got [0 10 10 10]
------- Stdout: -------
I170502 13:56:54.690054 46386 storage/store.go:1250  [n1,s1]: failed initial metrics computation: [n1,s1]: system config not yet available
I170502 13:56:54.690093 46386 gossip/gossip.go:297  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:59577" > attrs:<> locality:<> 
W170502 13:56:54.693215 46386 gossip/gossip.go:1189  [n2] no incoming or outgoing connections
I170502 13:56:54.693329 46386 storage/store.go:1250  [n2,s2]: failed initial metrics computation: [n2,s2]: system config not yet available
I170502 13:56:54.693361 46386 gossip/gossip.go:297  [n2] NodeDescriptor set to node_id:2 address:<network_field:"tcp" address_field:"127.0.0.1:51550" > attrs:<> locality:<> 
I170502 13:56:54.693497 46492 gossip/client.go:131  [n2] started gossip client to 127.0.0.1:59577
W170502 13:56:54.696114 46386 gossip/gossip.go:1189  [n3] no incoming or outgoing connections
I170502 13:56:54.696225 46386 storage/store.go:1250  [n3,s3]: failed initial metrics computation: [n3,s3]: system config not yet available
I170502 13:56:54.696273 46386 gossip/gossip.go:297  [n3] NodeDescriptor set to node_id:3 address:<network_field:"tcp" address_field:"127.0.0.1:41381" > attrs:<> locality:<> 
I170502 13:56:54.696299 46596 gossip/client.go:131  [n3] started gossip client to 127.0.0.1:59577
W170502 13:56:54.699626 46386 gossip/gossip.go:1189  [n4] no incoming or outgoing connections
I170502 13:56:54.699717 46386 storage/store.go:1250  [n4,s4]: failed initial metrics computation: [n4,s4]: system config not yet available
I170502 13:56:54.699743 46386 gossip/gossip.go:297  [n4] NodeDescriptor set to node_id:4 address:<network_field:"tcp" address_field:"127.0.0.1:45616" > attrs:<> locality:<> 
I170502 13:56:54.699905 46756 gossip/client.go:131  [n4] started gossip client to 127.0.0.1:59577
I170502 13:56:54.710630 46710 storage/replica_raftstorage.go:415  [s1,r1/1:/M{in-ax}] generated preemptive snapshot 82a442f8 at index 16
I170502 13:56:54.711036 46710 storage/store.go:3342  [s1,r1/1:/M{in-ax}] streamed snapshot to (n2,s2):?: kv pairs: 34, log entries: 6, rate-limit: 2.0 MiB/sec, 0ms
I170502 13:56:54.711259 46732 storage/replica_raftstorage.go:595  [s2,r1/?:{-}] applying preemptive snapshot at index 16 (id=82a442f8, encoded size=6059, 1 rocksdb batches, 6 log entries)
I170502 13:56:54.711608 46732 storage/replica_raftstorage.go:603  [s2,r1/?:/M{in-ax}] applied preemptive snapshot in 0ms [clear=0ms batch=0ms entries=0ms commit=0ms]
I170502 13:56:54.711889 46710 storage/replica_command.go:3537  [s1,r1/1:/M{in-ax}] change replicas (ADD_REPLICA (n2,s2):2): read existing descriptor r1:/M{in-ax} [(n1,s1):1, next=2]
I170502 13:56:54.712802 46736 storage/replica.go:2600  [s1,r1/1:/M{in-ax}] proposing ADD_REPLICA (n2,s2):2: [(n1,s1):1 (n2,s2):2]
I170502 13:56:54.713436 46899 storage/replica_raftstorage.go:415  [s1,r1/1:/M{in-ax}] generated preemptive snapshot 454e233e at index 18
I170502 13:56:54.714064 46899 storage/store.go:3342  [s1,r1/1:/M{in-ax}] streamed snapshot to (n3,s3):?: kv pairs: 37, log entries: 8, rate-limit: 2.0 MiB/sec, 1ms
I170502 13:56:54.714288 46766 storage/replica_raftstorage.go:595  [s3,r1/?:{-}] applying preemptive snapshot at index 18 (id=454e233e, encoded size=7619, 1 rocksdb batches, 8 log entries)
I170502 13:56:54.714648 46766 storage/replica_raftstorage.go:603  [s3,r1/?:/M{in-ax}] applied preemptive snapshot in 0ms [clear=0ms batch=0ms entries=0ms commit=0ms]
I170502 13:56:54.715004 46899 storage/replica_command.go:3537  [s1,r1/1:/M{in-ax}] change replicas (ADD_REPLICA (n3,s3):3): read existing descriptor r1:/M{in-ax} [(n1,s1):1, (n2,s2):2, next=3]
I170502 13:56:54.715823 46919 storage/raft_transport.go:436  raft transport stream to node 1 established
I170502 13:56:54.717437 46715 storage/replica.go:2600  [s1,r1/1:/M{in-ax}] proposing ADD_REPLICA (n3,s3):3: [(n1,s1):1 (n2,s2):2 (n3,s3):3]
I170502 13:56:54.718549 46896 storage/replica_raftstorage.go:415  [s1,r1/1:/M{in-ax}] generated preemptive snapshot 8b5bb8ee at index 20
I170502 13:56:54.719038 46896 storage/store.go:3342  [s1,r1/1:/M{in-ax}] streamed snapshot to (n4,s4):?: kv pairs: 40, log entries: 10, rate-limit: 2.0 MiB/sec, 0ms
I170502 13:56:54.719224 46910 storage/replica_raftstorage.go:595  [s4,r1/?:{-}] applying preemptive snapshot at index 20 (id=8b5bb8ee, encoded size=9243, 1 rocksdb batches, 10 log entries)
I170502 13:56:54.719629 46910 storage/replica_raftstorage.go:603  [s4,r1/?:/M{in-ax}] applied preemptive snapshot in 0ms [clear=0ms batch=0ms entries=0ms commit=0ms]
I170502 13:56:54.719968 46896 storage/replica_command.go:3537  [s1,r1/1:/M{in-ax}] change replicas (ADD_REPLICA (n4,s4):4): read existing descriptor r1:/M{in-ax} [(n1,s1):1, (n2,s2):2, (n3,s3):3, next=4]
I170502 13:56:54.721623 46912 storage/replica.go:2600  [s1,r1/1:/M{in-ax}] proposing ADD_REPLICA (n4,s4):4: [(n1,s1):1 (n2,s2):2 (n3,s3):3 (n4,s4):4]
I170502 13:56:54.863721 46954 storage/replica_command.go:3537  [s2,r1/2:/M{in-ax}] change replicas (REMOVE_REPLICA (n1,s1):1): read existing descriptor r1:/M{in-ax} [(n1,s1):1, (n2,s2):2, (n3,s3):3, (n4,s4):4, next=5]
I170502 13:56:54.865891 46960 storage/replica.go:2600  [s2,r1/2:/M{in-ax}] proposing REMOVE_REPLICA (n1,s1):1: [(n4,s4):4 (n2,s2):2 (n3,s3):3]
I170502 13:56:54.890595 46916 storage/store.go:3149  [s1,r1/1:/M{in-ax}] added to replica GC queue (peer suggestion)
I170502 13:56:54.890849 46091 storage/store.go:2137  [replicaGC,s1,r1/1:/M{in-ax}] removing replica
I170502 13:56:54.891030 46091 storage/replica.go:673  [replicaGC,s1,r1/1:/M{in-ax}] removed 34 (26+8) keys in 0ms [clear=0ms commit=0ms]
W170502 13:56:55.368059 46979 storage/replica.go:2274  [s2,r1/2:/M{in-ax}] context cancellation after 0.5s of attempting command Increment ["a",/Min)
I170502 13:57:25.140053 46401 gossip/gossip.go:1203  [n1] node has connected to cluster via gossip
I170502 13:57:29.067559 46782 gossip/gossip.go:1203  [n4] node has connected to cluster via gossip
I170502 13:57:31.334958 46516 gossip/gossip.go:1203  [n2] node has connected to cluster via gossip
I170502 13:57:31.778469 46630 gossip/gossip.go:1203  [n3] node has connected to cluster via gossip
I170502 13:57:40.590665 46502 vendor/google.golang.org/grpc/transport/http2_server.go:323  transport: http2Server.HandleStreams failed to read frame: read tcp 127.0.0.1:45616->127.0.0.1:53393: use of closed network connection
I170502 13:57:40.590706 46623 vendor/google.golang.org/grpc/transport/http2_server.go:323  transport: http2Server.HandleStreams failed to read frame: read tcp 127.0.0.1:41381->127.0.0.1:53148: use of closed network connection
I170502 13:57:40.590736 46499 vendor/google.golang.org/grpc/transport/http2_server.go:323  transport: http2Server.HandleStreams failed to read frame: read tcp 127.0.0.1:51550->127.0.0.1:45397: use of closed network connection
I170502 13:57:40.590793 46084 vendor/google.golang.org/grpc/transport/http2_server.go:323  transport: http2Server.HandleStreams failed to read frame: read tcp 127.0.0.1:59577->127.0.0.1:48379: use of closed network connection
W170502 13:57:40.590854 46871 storage/raft_transport.go:442  raft transport stream to node 3 failed: rpc error: code = Internal desc = transport is closing

```
Please assign, take a look and update the issue accordingly.

--- a/pkg/kv/dist_sender.go
+++ b/pkg/kv/dist_sender.go
@@ -1197,7 +1197,7 @@ func (ds *DistSender) sendToReplicas(
 			if err == nil {
 				// Determine whether the error must be propagated immediately or whether
 				// sending can continue to alternate replicas.
-				propogateError := false
+				propagateError := false
 				switch tErr := call.Reply.Error.GetDetail().(type) {
 				case nil:
 					return call.Reply, nil
@@ -1217,17 +1217,17 @@ func (ds *DistSender) sendToReplicas(
 							// Replace NotLeaseHolderError with RangeNotFoundError.
 							log.ErrEventf(ctx, "reported lease holder %s not in replicas slice %+v", lh, replicas)
 							call.Reply.Error = roachpb.NewError(roachpb.NewRangeNotFoundError(rangeID))
-							propogateError = true
+							propagateError = true
 						} else {
 							// Move the new lease holder to the head of the queue for the next retry.
 							transport.MoveToFront(*lh)
 						}
 					}
 				default:
-					propogateError = true
+					propagateError = true
 				}
 
-				if propogateError {
+				if propagateError {
 					// The error received is not specific to this replica, so we
 					// should return it instead of trying other replicas. However,
 					// if we're trying to commit a transaction and there are
@@ -1235,25 +1235,25 @@ func (ds *DistSender) sendToReplicas(
 					// was already received, we must return an ambiguous commit
 					// error instead of returned error.
 					log.ErrEventf(ctx, "application error: %s", call.Reply.Error)
-					if haveCommit {
-						timer := time.NewTimer(defaultPendingRPCTimeout)
-						defer timer.Stop()
-						// If there are still pending RPC(s), try to wait them out.
-						for timedOut := false; pending > 0 && !timedOut; {
-							select {
-							case pendingCall := <-done:
-								pending--
-								if err := pendingCall.Err; err != nil {
-									if grpc.Code(err) != codes.Unavailable {
-										ambiguousResult = true
-									}
-								} else if pendingCall.Reply.Error == nil {
-									return pendingCall.Reply, nil
+					timer := time.NewTimer(defaultPendingRPCTimeout)
+					defer timer.Stop()
+					// If there are still pending RPC(s), try to wait them out.
+					for timedOut := false; pending > 0 && !timedOut; {
+						select {
+						case pendingCall := <-done:
+							pending--
+							if err := pendingCall.Err; err != nil {
+								if grpc.Code(err) != codes.Unavailable {
+									ambiguousResult = true
 								}
-							case <-timer.C:
-								timedOut = true
+							} else if pendingCall.Reply.Error == nil {
+								return pendingCall.Reply, nil
 							}
+						case <-timer.C:
+							timedOut = true
 						}
+					}
+					if haveCommit {
 						if pending > 0 || ambiguousResult {
 							log.ErrEventf(ctx, "returning ambiguous result (pending=%d)", pending)
 							return nil, roachpb.NewAmbiguousResultError(

--- a/pkg/storage/client_test.go
+++ b/pkg/storage/client_test.go
@@ -513,7 +513,10 @@ func (t *multiTestContextKVTransport) SendNext(ctx context.Context, done chan<- 
 }
 
 func (t *multiTestContextKVTransport) NextReplica() roachpb.ReplicaDescriptor {
-	return roachpb.ReplicaDescriptor{}
+	if t.IsExhausted() {
+		return roachpb.ReplicaDescriptor{}
+	}
+	return t.replicas[t.idx].ReplicaDescriptor
 }
 
 func (t *multiTestContextKVTransport) MoveToFront(replica roachpb.ReplicaDescriptor) {

